### PR TITLE
Add configurable 'last' attribute in mapping rules

### PR DESCRIPTION
**Pull request by miguelsorianod**
Add the optional 'last' attribute in MappingRules for Products and Backends

### CODE REVIEW

File: pkg/controller/product/mapping_rules.go [hunk 1]
```diff
@@ -154,11 +166,17 @@ func (t *ThreescaleReconciler) createNewMappingRules(desiredList []capabilitiesv
...
+		if spec.Last != nil {
+			lastAttribute = *spec.Last
+		}
+
 		params := threescaleapi.Params{
 			"pattern":     spec.Pattern,
 			"http_method": spec.HTTPMethod,
 			"metric_id":   strconv.FormatInt(metricID, 10),
 			"delta":       strconv.Itoa(spec.Increment),
+			"last":        strconv.FormatBool(lastAttribute),
```
Comment (miguelsorianod) [inline pkg/controller/product/mapping_rules.go:179]: An alternative would be only specifying the 'last' parameter to 3scale api when it's been set to a non-nil value in the CR

**Review by eguzki — CHANGES_REQUESTED** [review]

File: pkg/controller/backend/3scale_reconciler.go [hunk 2]
```diff
@@ -475,6 +475,18 @@ func (t *ThreescaleReconciler) reconcileMatchedMappingRules(matchedList []mappin
...
 
+		//
+		// Reconcile last
+		//
+		desiredLastAttribute := false
+		if data.spec.Last != nil {
+			desiredLastAttribute = *data.spec.Last
+		}
+
+		if desiredLastAttribute != data.item.Last {
```
[the remark this answers is not in the document]
Comment (miguelsorianod) [reply pkg/controller/backend/3scale_reconciler.go:486]: I think if `data.spec.Last` is `nil` and `data.item.Last` is `true` we should set it to false as the default value when `data.spec.last` is `nil` would be `false` when not defined.
To me that it reconciliates only when is defined with a non-nil value in the CRD is confusing. CRD should be authoritative.

File: pkg/controller/product/mapping_rules.go (hunk 1 above)
Comment (miguelsorianod) [reply pkg/controller/product/mapping_rules.go]: Updated createNewMappingRules method to only send "last" parameter to the 3scale API when it is set to a non-nil value

File: pkg/controller/backend/3scale_reconciler.go [hunk 3]
```diff
@@ -498,11 +510,17 @@ func (t *ThreescaleReconciler) createNewMappingRules(desiredList []capabilitiesv
...
+		if spec.Last != nil {
+			lastAttribute = *spec.Last
+		}
+
 		params := threescaleapi.Params{
 			"pattern":     spec.Pattern,
 			"http_method": spec.HTTPMethod,
 			"metric_id":   strconv.FormatInt(metricID, 10),
 			"delta":       strconv.Itoa(spec.Increment),
+			"last":        strconv.FormatBool(lastAttribute),
```
[the remark this answers is not in the document]
Comment (miguelsorianod) [reply pkg/controller/backend/3scale_reconciler.go]: Updated createNewMappingRules method to only send "last" parameter to the 3scale API when it is set to a non-nil value in the CR

**Comment by eguzki** [comment]
missing doc

**Comment by miguelsorianod** [comment]
Added documentation

**Review by eguzki — APPROVED** [review]

**Pull request merged**